src/Tag.tsx

```diff
@@ -0,0 +1,178 @@
+import type { Equals } from "tsafe";
+import { symToStr } from "tsafe/symToStr";
+
+type DataAttribute = Record<`data-${string}`, string | boolean | null | undefined>;
+
+export type TagProps = TagProps.Common &
+    (TagProps.WithIcon | TagProps.WithoutIcon) &
+    (TagProps.AsAnchor | TagProps.AsButton | TagProps.AsSpan);
+export namespace TagProps {
+    export type Common = {
+        className?: string;
+        isSmall?: boolean;
+        style?: CSSProperties;
+        title?: string;
+        children: ReactNode;
+    };
+
+    export type WithIcon = {
+        /** Function of the button, to provide if the label isn't explicit */
+        iconId: FrIconClassName | RiIconClassName;
+    };
+
+    export type WithoutIcon = {
+        iconId?: never;
+    };
+
+    export type AsAnchor = {
+        linkProps: RegisteredLinkProps;
+        onClick?: never;
+        nativeButtonProps?: never;
+        nativeSpanProps?: never;
+        isDismissible?: never;
+        pressed?: never;
+    };
+    export type AsButton = {
+        linkProps?: never;
+        nativeSpanProps?: never;
+        isDismissible?: boolean;
+        pressed?: boolean;
+        onClick?: React.MouseEventHandler<HTMLButtonElement>;
+        nativeButtonProps?: React.DetailedHTMLProps<
+            React.ButtonHTMLAttributes<HTMLButtonElement>,
+            HTMLButtonElement
+        > &
+            DataAttribute;
+    };
+    export type AsSpan = {
+        linkProps?: never;
+        onClick?: never;
+        isDismissible?: never;
+        pressed?: never;
+        nativeButtonProps?: never;
+        nativeSpanProps?: React.DetailedHTMLProps<
+            React.HTMLAttributes<HTMLSpanElement>,
+            HTMLSpanElement
+        > &
+            DataAttribute;
+    };
+}
+
+/** @see <https://react-dsfr-components.etalab.studio/?path=/docs/components-tag> */
+export const Tag = memo(
+    forwardRef<HTMLButtonElement | HTMLAnchorElement | HTMLSpanElement, TagProps>((props, ref) => {
+        const {
+            className: prop_className,
+            children,
+            title,
+            iconId,
+            isSmall,
+            pressed,
+            isDismissible,
+            linkProps,
+            nativeButtonProps,
+            nativeSpanProps,
+            style,
+            onClick,
+            ...rest
+        } = props;
+
+        assert<Equals<keyof typeof rest, never>>();
+
+        const { Link } = getLink();
+
+        const className = cx(
+            fr.cx(
+                "fr-tag",
+                isSmall && `fr-tag--sm`,
+                iconId,
+                iconId && "fr-tag--icon-left", // actually, it's always left but we need it in order to have the icon rendering
+                isDismissible && "fr-tag--dismiss"
+            ),
+            linkProps !== undefined && linkProps.className,
+            prop_className
+        );
+
+        return (
+            <>
+                {linkProps && (
```

Comment:
```suggestion
                {linkProps !== undefined && (
```
We never use non boolean value as if they where boolean, it makes the code harder to reason about and it's the source of countless bugs.